src/server/scripts/EasternKingdoms/Stratholme/boss_maleki_the_pallid.cpp

```diff
@@ -87,32 +87,37 @@ class boss_maleki_the_pallid : public CreatureScript
             if (!UpdateVictim())
                 return;
 
-            //Frostbolt
-            if (Frostbolt_Timer <= diff)
-            {
-                if (rand32() % 100 < 90)
-                    DoCastVictim(SPELL_FROSTBOLT);
-                Frostbolt_Timer = 3500;
-            } else Frostbolt_Timer -= diff;
-
-            //IceTomb
-            if (IceTomb_Timer <= diff)
-            {
-                if (rand32() % 100 < 65)
-                    DoCastVictim(SPELL_ICETOMB);
-                IceTomb_Timer = 28000;
-            } else IceTomb_Timer -= diff;
+            _events.Update(diff);
 
-            //DrainLife
-            if (DrainLife_Timer <= diff)
+            while (uint32 eventId = _events.ExecuteEvent())
             {
-                if (rand32() % 100 < 55)
-                    DoCastVictim(SPELL_DRAINLIFE);
-                DrainLife_Timer = 31000;
-            } else DrainLife_Timer -= diff;
+                switch (eventId)
+                {
+                    case EVENT_FROSTBOLT:
+                        if (rand() % 100 < 90)
```

Comment:
simply create a commit in your 3.3.5 branch

pull requests are nothing but branches, whatever changes you do to the branch you want to PR (in this case, `3.3.5`) will be reflected in this PR